src/main/java/au/com/codeka/carrot/bindings/JsonArrayBindings.java

```diff
@@ -0,0 +1,67 @@
+package au.com.codeka.carrot.bindings;
+
+import au.com.codeka.carrot.Bindings;
+import au.com.codeka.carrot.ValueHelper;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+
+/**
+ * {@link Bindings} based on the values in a {@link JSONArray}.
+ *
+ * @author Marten Gajda
+ */
+public final class JsonArrayBindings implements Bindings, Iterable
```

Comment:
I was considering that. My idea was to give a brief example of how to create adapters for arbitrary sources.

The json.org dependency is added as [`compileOnly`](https://blog.gradle.org/introducing-compile-only-dependencies) which is the equivalent to Maven's `provided`. As a result, this library won't pull in `json.org`<sup>[1](#footnote)</sup> and you can not use this adapter unless you added the `json.org` dependency in your own project which uses carrot.

Does that make sense?

That being said I don't mind creating a sub-module if you prefer.


<a name="footnote">[1]</a>  see the pom which was created by jitpack for my branch: [carrot-stories~3-Bindings-v2.2.0-g2e98436-8.pom](https://jitpack.io/com/github/dmfs/carrot/stories~3-Bindings-v2.2.0-g2e98436-8/carrot-stories~3-Bindings-v2.2.0-g2e98436-8.pom), it contains `json.org` only as a test dependency